frontend/app/.server/routes/helpers/protected-renew-route-helpers.ts

```diff
@@ -353,29 +356,33 @@ export function loadProtectedRenewStateForReview({ params, session }: LoadProtec
   }
 }
 
-export function isPrimaryApplicantStateComplete(state: ProtectedRenewState) {
-  return state.dentalInsurance !== undefined && state.demographicSurvey !== undefined;
+export function isPrimaryApplicantStateComplete(state: ProtectedRenewState, appContainer: AppContainerProvider) {
+  const { ENABLED_FEATURES } = appContainer.get(TYPES.configs.ClientConfig);
+  const demographicSurveyEnabled = ENABLED_FEATURES.includes('demographic-survey');
+  return state.dentalInsurance !== undefined && (demographicSurveyEnabled ? state.demographicSurvey !== undefined : true);
 }
 
-export function isChildrenStateComplete(state: ProtectedRenewState) {
-  return state.children.every((child) => child.isParentOrLegalGuardian !== undefined && child.dentalInsurance !== undefined && child.demographicSurvey !== undefined);
+export function isChildrenStateComplete(state: ProtectedRenewState, appContainer: AppContainerProvider) {
+  const { ENABLED_FEATURES } = appContainer.get(TYPES.configs.ClientConfig);
+  const demographicSurveyEnabled = ENABLED_FEATURES.includes('demographic-survey');
+  return state.children.every((child) => child.isParentOrLegalGuardian !== undefined && child.dentalInsurance !== undefined && (demographicSurveyEnabled ? child.demographicSurvey !== undefined : true));
```

Comment:
Consider modifying these functions to accept `demographicSurveyEnabled` as a `boolean` rather than `appContainer`. This change will make this function easier to test because we wouldn't need to mock or instantiate `appContainer` - we can just pass a `boolean` value directly. This change would also isolate this function from dependency injection concerns.